belvedere/src/main/java/zendesk/belvedere/BelvedereUi.java

```diff
@@ -210,9 +212,29 @@ public void run() {
 
                 @Override
                 public void onPermissionsDenied() {
-                    Activity appCompatActivity = popupBackend.getActivity();
-                    if(appCompatActivity != null) {
-                        Toast.makeText(appCompatActivity, R.string.belvedere_permissions_denied, Toast.LENGTH_SHORT).show();
+                    final Activity appCompatActivity = popupBackend.getActivity();
+                    if (appCompatActivity != null) {
+                        final ViewGroup parentView = appCompatActivity.findViewById(android.R.id.content);
+                        Utils.showSnackBar(
+                                parentView,
+                                appCompatActivity.getString(R.string.belvedere_permissions_denied),
+                                Snackbar.LENGTH_LONG,
+                                appCompatActivity.getString(R.string.belvedere_navigate_to_settings),
+                                new OnClickListener() {
+                                    @Override
+                                    public void onClick(View v) {
+                                        Intent settingsIntent = new Intent();
+                                        settingsIntent.setAction(Settings.ACTION_APPLICATION_DETAILS_SETTINGS);
+                                        settingsIntent.setData(
+                                                Uri.fromParts(
+                                                        "package",
```

Comment:
Just a small nit pick. Can we define a constant for this? It seems this class already defined a few constants already.

Reply:
Nice catch 👌🏻